app/controllers/time_clock_punches_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TimeClockPunchesController < InternalController
+  before_action :set_time_clock_punch, only: %i[ show edit update destroy ]
+  before_action :set_time_clock_period_scope, only: %i[ new create edit update ]
+
+  # GET /time_clock_punches
+  def index
+    @q = policy_scope(TimeClockPunch).ransack(params[:q])
+    @time_clock_punches = @q.result(distinct: true).page(params[:page])
```

Comment:
These should have an explicit order - probably by start time & end time.

Reply:
Done